In [13]:
# Loading Libraries
import pandas as pd
import plotly_express as px
import matplotlib.pyplot as plt
import numpy as np

%matplotlib inline
plt.rcParams['figure.figsize']=(10,8)

In [ ]:
# School Id

In [23]:
# Loading Data
df = pd.read_csv('user_educations.csv')
df.shape, df.head()

((61295, 3),
    user_id  school_id  degree_id
 0      151    11825.0        2.0
 1      151    13556.0        NaN
 2       41    11929.0        1.0
 3       41    13087.0        NaN
 4       51   999999.0        NaN)

In [25]:
df.dtypes

user_id      object
school_id    object
degree_id    object
dtype: object

In [24]:
# Fixing type variables
type_correction = {'user_id': 'object',
                  'school_id': 'object',
                  'degree_id': 'object'}
df = df.astype(type_correction)

In [26]:
# Missing Value Proportions
np.round(df.isnull().sum()/df.shape[0] * 100, 2)

user_id       0.00
school_id     0.00
degree_id    99.13
dtype: float64

In [27]:
# Count of Uniuqe Categories
df.nunique()

user_id      41139
school_id     1444
degree_id        3
dtype: int64

In [42]:
# Univariate Analysis of school_id
temp = df.school_id.value_counts()
temp.head()

999999.0    15275
11098.0      2239
11278.0      1519
10329.0      1295
13370.0       912
Name: school_id, dtype: int64

In [52]:
bins = [0, 50, 100, 200, 500, 1000, 5000, 10000, 50000]
pd.cut(temp, bins=bins)

999999.0    (10000, 50000]
11098.0       (1000, 5000]
11278.0       (1000, 5000]
10329.0       (1000, 5000]
13370.0        (500, 1000]
                 ...      
2762.0             (0, 50]
10957.0            (0, 50]
7072.0             (0, 50]
10975.0            (0, 50]
8193.0             (0, 50]
Name: school_id, Length: 1444, dtype: category
Categories (8, interval[int64]): [(0, 50] < (50, 100] < (100, 200] < (200, 500] < (500, 1000] < (1000, 5000] < (5000, 10000] < (10000, 50000]]

# Insights
- Fix type for `school_id` & `degree_id` to `object`.
- Drop `degree_id`, it has missing data for 99% for  observations
- Data avail for `41139` user only
- There are `1444` unique schools in the region for study.